crates/common/types/transaction.rs

```diff
@@ -35,6 +37,116 @@ pub enum Transaction {
     PrivilegedL2Transaction(PrivilegedL2Transaction),
 }
 
+/// The same as a Transaction enum, only that blob transactions are in wrapped format, including
+/// the blobs bundle.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum P2PTransaction {
+    LegacyTransaction(LegacyTransaction),
+    EIP2930Transaction(EIP2930Transaction),
+    EIP1559Transaction(EIP1559Transaction),
+    WrappedEIP4844Transaction(WrappedEIP4844Transaction),
+    PrivilegedL2Transaction(PrivilegedL2Transaction),
+}
+
+impl TryInto<Transaction> for P2PTransaction {
+    type Error = String;
+
+    fn try_into(self) -> Result<Transaction, Self::Error> {
+        match self {
+            P2PTransaction::LegacyTransaction(itx) => Ok(Transaction::LegacyTransaction(itx)),
+            P2PTransaction::EIP2930Transaction(itx) => Ok(Transaction::EIP2930Transaction(itx)),
+            P2PTransaction::EIP1559Transaction(itx) => Ok(Transaction::EIP1559Transaction(itx)),
+            P2PTransaction::PrivilegedL2Transaction(itx) => {
+                Ok(Transaction::PrivilegedL2Transaction(itx))
+            }
+            _ => Err("Can't convert blob p2p transaction into regular transaction. Blob bundle would be lost.".to_string()),
+        }
+    }
+}
+
+impl RLPEncode for P2PTransaction {
+    fn encode(&self, buf: &mut dyn bytes::BufMut) {
+        match self {
+            P2PTransaction::LegacyTransaction(t) => t.encode(buf),
+            tx => Bytes::copy_from_slice(&tx.encode_canonical_to_vec()).encode(buf),
+        };
+    }
+}
+
+impl RLPDecode for P2PTransaction {
+    fn decode_unfinished(rlp: &[u8]) -> Result<(Self, &[u8]), RLPDecodeError> {
+        if is_encoded_as_bytes(rlp)? {
+            // Adjust the encoding to get the payload
+            let payload = get_rlp_bytes_item_payload(rlp);
+            let tx_type = payload.first().unwrap();
```

Comment:
I agree that if we're in inside this if branch, then this should not panic, but invariants change with time. It might be overkill, but why don't we map this option to an error if  tx_type is none?